src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -744,26 +744,28 @@ protected String getPodNameSpace(Pod pod) {
             return pod.getMetadata().getNamespace();
         }
 
-        private SlaveOperationDetails waitForSlaveToComeOnline(String podId, String namespace, String slaveName) throws Exception {
-            Slave slave = (Slave) jenkins().getNode(slaveName);
-            Pod pod = getPodByNamespaceAndPodId(namespace, podId);
-            int slaveConnectTimeoutInSeconds = t.getSlaveConnectTimeout();
-            int attemptsToMake = slaveConnectTimeoutInSeconds / SlaveTimeLimitedTaskRunner.DEFAULT_SLEEP_AFTER_FAILURE_IN_MILLIS;
+        @VisibleForTesting
+        SlaveOperationDetails waitForSlaveToComeOnline(String podId, String namespace) throws Exception {
+            // Default delay is 1 second, so max number of attempts == number of seconds
+            int attemptsToMake = t.getSlaveConnectTimeout();
```

Comment:
This still sounds strange. Assignment is valid only with assumption that default delay wasn't change. Can't we bring in delay value here in attempts calculations?